courseScheduleII.js

```diff
@@ -0,0 +1,113 @@
+/*
+For this approach, we can represent the input as a dependency graph where in order to take a course, 
+you have to complete a prereq, which is the same as two nodes connected by an edge.
+
+Since we are trying to find a valid ordering for all these courses that respects our prereqs,
+we can approach this problem with a topological sort algorithm where if there happens to be an
+edge between a prereq and a class, it means that the prereq appears before the class in a linear ordering fashion.
+
+A class that has a prereq before would be an indegree relationship and we can the can use our topological sort algorithm to track
```

Comment:
These are implementation details, but you're not describing what this algorithm is doing conceptually.